storage/src/state/ledger.rs

```diff
@@ -1285,8 +1286,14 @@ impl<N: Network> BlockState<N> {
 
     /// Returns the block height for the given block hash.
     fn get_block_height(&self, block_hash: &N::BlockHash) -> Result<u32> {
-        match self.block_headers.get(block_hash)? {
-            Some(block_header) => Ok(block_header.height()),
+        // Performance note: this method is using `Storage::get_raw` in order to avoid costly
+        // (and duplicate) proof verification, which happens automatically upon full deserialization.
+        match self.block_headers.get_raw(block_hash)? {
+            Some(raw_block_header) => {
+                let raw_block_height: [u8; 4] = raw_block_header[64..][..4].try_into().unwrap(); // safe - we stored the headers ourself
```

Comment:
My worry here is that if we change the serialization of block headers for mainnet (will likely add additional fields), this will be left unchecked.

Reply:
Hmm, how about if I deserialize `HeaderMetadata`? I'll see how this can be guarded against better :+1: .

Reply:
Actually, there's access to `Network` there, so we can assert this is the expected `NETWORK_ID`.

Reply:
I think the NETWORK_ID flag should suffice for now. Once we have additional network ids here, we can look into adding a match case for it.

Reply:
Currently rethinking the approach of using `get_raw`. 

I don't believe dealing with bytes directly is the approach we want to move forward with. It introduces potential issues down the line and also doesn't sit well with the abstractions we currently have.

Reply:
We are always dealing with raw bytes under the hood in our storage; as long as deserialization is computationally heavy, we need to find ways to avoid it in places where it's just not needed.